bundles/org.openhab.ui.dashboard/src/main/java/org/openhab/ui/dashboard/internal/DashboardService.java

```diff
@@ -202,4 +226,34 @@ private void addTilesForExternalServices(Map<String, Object> properties) {
             }
         }
     }
+
+    /**
+     * Get a localized text considering the current locale (language)
+     *
+     * @param key the key starting with &#64;text/ of to the localization string, e.g &#64;text/index.welcome
+     *
+     * @return the localized text if the key is found, or an empty string if not
```

Comment:
"or an empty string if not" 😲 
This can't be the solution. Strings will be added over time and if the Japanese language file is not updated right away, please present the english string as a fallback.

Reply:
Yes or course the English text is returned for a key that is not present in the Japanese file.
Empty string is returned in the case the entry is not defined in the default English file too. This case should never happened except if there is a bug in the code.